src/js/utils.js

```diff
@@ -3,6 +3,9 @@ export const setCookie = (value) => {
   d.setTime(d.getTime() + 24 * 60 * 60 * 1000);
   const expires = 'expires=' + d.toUTCString();
   document.cookie = '_cookies_accepted=' + value + '; ' + expires;
+  if (value == 'all' || value == 'functionality') {
```

Comment:
We should check if dataLayer exists here too.

Reply:
Isn't site tracking attached to performance instead of functionality?

Reply:
I am using performance for GA

Reply:
If so we shouldn't track people that only opt for functionality.

Reply:
re the datalayer, we can create it, if it doesn't exist.

Reply:
Yeah but if you create it `dataLayer.push({event: 'pageview'});` does nothing but looks from the code it should work. I think it's better to gate the call on the existence of the object.

Reply:
I guess there won't be a race conditon?